### PR TITLE
Use next/link for components filter

### LINKED ISSUE #5636: [Polaris website refresh] Use next.js Link component for Component filters
Right now, they're using `router.push()` which means the pages aren't preloaded.

(This is a bit of a head scratcher: On the one hand, preloading full pages is a bit silly, since we're just doing a `components.filter()` to narrow the components index down. However, we also want proper URLs for each component category, so maybe full pages is the best option after all.)

### PR DESCRIPTION
**Pull request by martenbjork**
Fixes #5636

### CODE REVIEW

File: polaris.shopify.com/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 
-interface Props extends HTMLProps<HTMLButtonElement> {
+interface Props {
   small?: boolean;
   pill?: boolean;
-  children: React.ReactNode;
 }
 
-function Button({ small, pill, children, ...rest }: Props) {
+interface NativeButtonProps extends Props, HTMLProps<HTMLButtonElement> {}
```
Comment (aaronccasanova) [inline polaris.shopify.com/src/components/Button/Button.tsx]:
```suggestion
interface ButtonProps extends Props, HTMLProps<HTMLButtonElement> {}
```

Light suggestion to keep prop interfaces consistent with the associated component.

Comment (martenbjork) [reply polaris.shopify.com/src/components/Button/Button.tsx]: You mean the name of the interface? (Definitely agree!)

File: polaris.shopify.com/src/components/Button/Button.tsx [hunk 2]
```diff
@@ -23,4 +26,27 @@ function Button({ small, pill, children, ...rest }: Props) {
...
+  ...rest
+}: LinkButtonProps) {
+  return (
+    <Link href={href} passHref>
+      <a
+        className={[
+          styles.Button,
+          small ? styles.small : null,
+          pill ? styles.pill : null,
+        ].join(" ")}
```
Comment (aaronccasanova) [inline polaris.shopify.com/src/components/Button/Button.tsx]:
```suggestion
        className={[
          styles.Button,
          small ? styles.small : null,
          pill ? styles.pill : null,
        ].filter(Boolean).join(" ")}
```

**suggestion (non-blocking):** Not a show stopper but we should probably filter out the `null` values
image

**Review by aaronccasanova — APPROVED** [review]

File: polaris.shopify.com/src/components/Button/Button.tsx [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
   small?: boolean;
   pill?: boolean;
-  children: React.ReactNode;
 }
 
-function Button({ small, pill, children, ...rest }: Props) {
+interface ButtonProps extends Props, HTMLProps<HTMLButtonElement> {}
+interface LinkButtonProps extends Props, PropsWithChildren<LinkProps> {}
+
+function Button({ small, pill, children, ...rest }: NativeButtonProps) {
```
Comment (aaronccasanova) [inline polaris.shopify.com/src/components/Button/Button.tsx]:
```suggestion
function Button({ small, pill, children, ...rest }: ButtonProps) {
```

File: polaris.shopify.com/src/components/Button/Button.tsx (hunk 2 above)
Comment (martenbjork) [reply polaris.shopify.com/src/components/Button/Button.tsx]: Good call! I introduced a `className` helper function to do this automatically.

**Pull request merged**